src/Raven.Server/Documents/Sharding/Queries/ShardedQueryProcessor.cs

```diff
@@ -115,7 +115,7 @@ public void Initialize(out Dictionary<int, BlittableJsonReaderObject> queryTempl
         queryTemplates = new(_parent.DatabaseContext.ShardCount);
         for (int i = 0; i < _parent.DatabaseContext.ShardCount; i++)
         {
-            queryTemplates.Add(i, queryTemplate.Clone(_context));
```

Comment:
Why do we need to have it per shard? Cannot we just use single `queryTemplate` instance?

Reply:
`GenerateLoadByIdQueries()` creates a different queryTemplate for each shard

Reply:
Oh, okay. Makes sense then